### ISSUE TITLE
How are model parameters updated in the outerloop?

### ISSUE THREAD

**Issue by timbmg** [tristandeleu/pytorch-maml#17]
Hi @tristandeleu, thanks for the awesome library!

I dont quite understand how the original parameters of the model are updated. I am specifically referring to these lines in maml.py:
```python3
# inner loop
params, adaptation_results = self.adapt(...)
...
# outer loop
test_logits = self.model(test_inputs, params=params)
```
From my understanding that means that in the outer loop also the original parameters are not used, since we are telling the model to use the adapted params. Therefore I am wondering where the gradients for the parmas in the model are coming from?

Would be great if you could help me out here.

**Comment by timbmg** [comment]
nevermind, i got it.

For anyone who might find this, the graph is built inside `gradient_update_parameters` when `params = None`, then the original params are used.
